path/join_test.ts

```diff
@@ -56,10 +58,14 @@ const joinTests =
     [["/", "", "/foo"], "/foo"],
     [["", "/", "foo"], "/foo"],
     [["", "/", "/foo"], "/foo"],
+
+    // URLs
+    [[new URL("file:///"), "x/b", "..", "/b/c.js"], "/x/b/c.js"],
+    [[new URL("file:///foo"), "../../../bar"], "/bar"],
```

Comment:
The first test case looks good. But I'd expect the result of the 2nd test case to be `"/../../bar"`.

Reply:
`/bar` is the normalized form of `/../../bar`. There's no parent directory of the root directory

Reply:
Oh, I tested with `node:path` and that does appear to be the case. That doesn't really make sense to me but I guess is fine 🤷🏾‍♂️

Reply:
How did you test it? The below prints `/bar` both in Node.js and Deno

```js
import { join } from "node:path";
console.log(join("/foo", "../../../bar"));
```

BTW `join("file:///foo", "../../../bar")` becomes `../bar` because `"file:///foo"` is considered relative path `file:/bar` (`file:` part is considered as directory name instead of scheme name)